### PR TITLE
Adding fabricator and generex

### PR DESCRIPTION
**Pull request by ramindu90**

### CODE REVIEW

File: generex/1.0.0-wso2v1/pom.xml [hunk 1]
```diff
@@ -0,0 +1,61 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+
+    <modelVersion>4.0.0</modelVersion>
+    <groupId>com.github.mifmif.wso2</groupId>
```
Comment (cnapagoda) [inline generex/1.0.0-wso2v1/pom.xml]: Group is should start with org.wso2.orbit

File: fabricator/2.1.4-wso2v1/pom.xml [hunk 2]
```diff
@@ -0,0 +1,61 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+
+    <modelVersion>4.0.0</modelVersion>
+    <groupId>com.github.azakordonets.wso2</groupId>
```
Comment (cnapagoda) [inline fabricator/2.1.4-wso2v1/pom.xml]: Group is should start with org.wso2.orbit

File: fabricator/2.1.4-wso2v1/pom.xml [hunk 3]
```diff
@@ -0,0 +1,61 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (cnapagoda) [inline fabricator/2.1.4-wso2v1/pom.xml]: Missing licence headers

File: generex/1.0.0-wso2v1/pom.xml [hunk 4]
```diff
@@ -0,0 +1,61 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
```
Comment (cnapagoda) [inline generex/1.0.0-wso2v1/pom.xml]: Missing license headers

File: fabricator/2.1.4-wso2v1/pom.xml [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    <url>http://wso2.org</url>
+
+    <dependencies>
+        <dependency>
+            <groupId>com.github.azakordonets</groupId>
+            <artifactId>fabricator_2.10</artifactId>
+            <version>${fabricator.version}</version>
+        </dependency>
+    </dependencies>
+
```
Comment (cnapagoda) [inline fabricator/2.1.4-wso2v1/pom.xml]: Missing distribution management section

File: generex/1.0.0-wso2v1/pom.xml [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+    <url>http://wso2.org</url>
+
+    <dependencies>
+        <dependency>
+            <groupId>com.github.mifmif</groupId>
+            <artifactId>generex</artifactId>
+            <version>${generex.version}</version>
+        </dependency>
+    </dependencies>
+
```
Comment (cnapagoda) [inline generex/1.0.0-wso2v1/pom.xml]: Missing distribution management section

File: generex/1.0.0-wso2v1/pom.xml [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+    </build>
+
+    <properties>
+        <generex.version>1.0.0</generex.version>
+        <maven.bundle.plugin.version>2.4.0</maven.bundle.plugin.version>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    </properties>
+
+
+</project>
```
Comment (cnapagoda) [inline generex/1.0.0-wso2v1/pom.xml]: No new line at end of the file

File: fabricator/2.1.4-wso2v1/pom.xml [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+    <properties>
+        <fabricator.version>2.1.4</fabricator.version>
+        <maven.bundle.plugin.version>2.4.0</maven.bundle.plugin.version>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    </properties>
+
+
+
+</project>
```
Comment (cnapagoda) [inline fabricator/2.1.4-wso2v1/pom.xml]: No new line at end of the file

File: fabricator/2.1.4-wso2v1/pom.xml [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+        </plugins>
+    </build>
+
+    <properties>
+        <fabricator.version>2.1.4</fabricator.version>
+        <maven.bundle.plugin.version>2.4.0</maven.bundle.plugin.version>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    </properties>
+
+
```
Comment (cnapagoda) [inline fabricator/2.1.4-wso2v1/pom.xml]: unwanted  spaces

File: fabricator/2.1.4-wso2v1/pom.xml (hunk 3 above)
Comment (ramindu90) [inline fabricator/2.1.4-wso2v1/pom.xml]: @cnapagoda added the requested changes. Please review and merge

File: fabricator/2.1.4-wso2v1/pom.xml (hunk 5 above)
Comment (ramindu90) [inline fabricator/2.1.4-wso2v1/pom.xml]: @cnapagoda added the requested changes. Please review and merge

File: generex/1.0.0-wso2v1/pom.xml (hunk 6 above)
Comment (ramindu90) [inline generex/1.0.0-wso2v1/pom.xml]: @cnapagoda added the requested changes. Please review and merge

File: fabricator/2.1.4-wso2v1/pom.xml [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+                        <Bundle-SymbolicName>${project.artifactId}</Bundle-SymbolicName>
+                        <Bundle-Name>${project.artifactId}</Bundle-Name>
+                        <Export-Package>
+                            fabricator.*;-split-package:=merge-first,
+                        </Export-Package>
+                        <Private-Package>
+                            *;-split-package:=merge-first
+                        </Private-Package>
+                        <Import-Package>
+                            *;resolution:=optional
```
Comment (cnapagoda) [inline fabricator/2.1.4-wso2v1/pom.xml]: We are not recommending to have optional imports

File: fabricator/2.1.4-wso2v1/pom.xml [hunk 11]
```diff
@@ -0,0 +1,90 @@
...
+    <description>
+        This bundle will export packages from Fabricator
+    </description>
+    <url>http://wso2.org</url>
+
+    <dependencies>
+        <dependency>
+            <groupId>com.github.azakordonets</groupId>
+            <artifactId>fabricator_2.10</artifactId>
+            <version>${fabricator.version}</version>
```
Comment (thusithathilina) [inline fabricator/2.1.4-wso2v1/pom.xml]: Dependencies should be optional

File: fabricator/2.1.4-wso2v1/pom.xml [hunk 12]
```diff
@@ -0,0 +1,90 @@
...
+                <groupId>org.apache.felix</groupId>
+                <artifactId>maven-bundle-plugin</artifactId>
+                <version>${maven.bundle.plugin.version}</version>
+                <extensions>true</extensions>
+                <configuration>
+                    <instructions>
+                        <Bundle-SymbolicName>${project.artifactId}</Bundle-SymbolicName>
+                        <Bundle-Name>${project.artifactId}</Bundle-Name>
+                        <Export-Package>
+                            fabricator.*;-split-package:=merge-first,
```
Comment (thusithathilina) [inline fabricator/2.1.4-wso2v1/pom.xml]: Need export versions

File: fabricator/2.1.4-wso2v1/pom.xml (hunk 10 above)
Comment (thusithathilina) [inline fabricator/2.1.4-wso2v1/pom.xml]: Plus can we have exact imports?

File: fabricator/2.1.4-wso2v1/pom.xml [hunk 13]
```diff
@@ -0,0 +1,90 @@
...
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    </properties>
+
+    <distributionManagement>
+        <repository>
+            <id>wso2.releases</id>
+            <name>WSO2 internal Repository</name>
+            <url>http://maven.wso2.org/nexus/content/repositories/releases/</url>
+        </repository>
+        <snapshotRepository>
```
Comment (thusithathilina) [inline fabricator/2.1.4-wso2v1/pom.xml]: No need of snapshot repo

File: generex/1.0.0-wso2v1/pom.xml [hunk 14]
```diff
@@ -0,0 +1,91 @@
...
+    <description>
+        This bundle will export packages from Generex
+    </description>
+    <url>http://wso2.org</url>
+
+    <dependencies>
+        <dependency>
+            <groupId>com.github.mifmif</groupId>
+            <artifactId>generex</artifactId>
+            <version>${generex.version}</version>
```
Comment (thusithathilina) [inline generex/1.0.0-wso2v1/pom.xml]: dependencies should be optional

File: generex/1.0.0-wso2v1/pom.xml [hunk 15]
```diff
@@ -0,0 +1,91 @@
...
+                <groupId>org.apache.felix</groupId>
+                <artifactId>maven-bundle-plugin</artifactId>
+                <version>${maven.bundle.plugin.version}</version>
+                <extensions>true</extensions>
+                <configuration>
+                    <instructions>
+                        <Bundle-SymbolicName>${project.artifactId}</Bundle-SymbolicName>
+                        <Bundle-Name>${project.artifactId}</Bundle-Name>
+                        <Export-Package>
+                            com.mifmif.common.regex.*;-split-package:=merge-first,
```
Comment (thusithathilina) [inline generex/1.0.0-wso2v1/pom.xml]: what is the export version?

File: generex/1.0.0-wso2v1/pom.xml [hunk 16]
```diff
@@ -0,0 +1,91 @@
...
+                        </Export-Package>
+                        <Private-Package>
+                            *;-split-package:=merge-first
+                        </Private-Package>
+                        <Import-Package>
+                            *;resolution:=optional
```
Comment (thusithathilina) [inline generex/1.0.0-wso2v1/pom.xml]: Better to put exact imports

File: generex/1.0.0-wso2v1/pom.xml [hunk 17]
```diff
@@ -0,0 +1,91 @@
...
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    </properties>
+
+    <distributionManagement>
+        <repository>
+            <id>wso2.releases</id>
+            <name>WSO2 internal Repository</name>
+            <url>http://maven.wso2.org/nexus/content/repositories/releases/</url>
+        </repository>
+        <snapshotRepository>
```
Comment (thusithathilina) [inline generex/1.0.0-wso2v1/pom.xml]: No need of snapshot

File: pom.xml [hunk 18]
```diff
@@ -64,6 +64,8 @@
         <module>compass/4.3.0.wso2v1</module>
         <module>poi/4.3.0.wso2v1</module>
```
Comment (thusithathilina) [inline pom.xml]: no need to add to parent pom

File: fabricator/2.1.4-wso2v1/pom.xml (hunk 11 above)
Comment (ramindu90) [inline fabricator/2.1.4-wso2v1/pom.xml]: @thusithathilina added the requested changes. Please review and merge

File: generex/1.0.0-wso2v1/pom.xml (hunk 14 above)
Comment (ramindu90) [inline generex/1.0.0-wso2v1/pom.xml]: @thusithathilina added the requested changes. Please review and merge

File: pom.xml (hunk 18 above)
Comment (ramindu90) [inline pom.xml]: @thusithathilina added the requested changes. Please review and merge

**Pull request closed without merging**